lib/excon/middlewares/idempotent.rb

```diff
@@ -20,6 +20,9 @@ def error_call(datum)
 
         if datum[:idempotent] && [Excon::Errors::Timeout, Excon::Errors::SocketError,
             Excon::Errors::HTTPStatusError].any? {|ex| datum[:error].kind_of?(ex) } && datum[:retries_remaining] > 1
+
+          sleep(datum[:retry_interval]) if datum[:retry_interval] && !datum[:mock]
```

Comment:
Yeah, removing it and setting it to 0 in tests seems reasonable for now. stubbing sleep and counting would probably be even better, but I don't think we have to do it now.